### PR TITLE
Add new `check_private_items` config

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes https://github.com/rust-lang/rust-clippy/issues/11742.

It adds a new configuration which allows for `MISSING_SAFETY_DOC`, `UNNECESSARY_SAFETY_DOC`, `MISSING_PANICS_DOC` and `MISSING_ERRORS_DOC` lints to run on private items.

changelog: [`missing_safety_doc`], [`unnecessary_safety_doc`], [`missing_panics_doc`], [`missing_errors_doc`]: Added the [`check-private-items`] configuration to enable lints on private items.
[#11842](https://github.com/rust-lang/rust-clippy/pull/11842)

r? @blyxyas

### CODE REVIEW

**Comment by xFrednet** [comment]
If this config is specific to doc lints it might be good to include that in the config name. If it's intended as a general setting, then the name should be fine :)

File: clippy_config/src/conf.rs [hunk 1]
```diff
@@ -543,6 +543,10 @@ define_Conf! {
     /// for _ in &mut *rmvec {}
     /// ```
     (enforce_iter_loop_reborrow: bool = false),
+    /// Lint: MISSING_SAFETY_DOC, UNNECESSARY_SAFETY_DOC, MISSING_PANICS_DOC, MISSING_ERRORS_DOC
+    ///
+    /// Whether to also run the listed lints on private items.
+    (check_private_items: bool = false),
```
Comment (blyxyas) [inline clippy_config/src/conf.rs:549]:
```suggestion
    (docs_check_private_items: bool = false),
```

**Review by blyxyas — CHANGES_REQUESTED** [review]
Meow Meow
Currently blocked by #11801

**Comment by GuillaumeGomez** [comment]
I kinda had in mind something that could be generalized indeed. Should I make the change to make it specific for docs?

**Review by blyxyas — APPROVED** [review]
Hmmm, you're probably right, it should be a general configuration option. Thanks for the clarification.

LGTM, thanks! :heart:

**Comment by GuillaumeGomez** [comment]
Great! I blessed the failing test so now waiting for the other PR to get merged. :)

**Comment by GuillaumeGomez** [comment]
Rebased and fixed the merge conflict.

**Review by blyxyas — APPROVED** [review]
Meow

**Comment by GuillaumeGomez** [comment]
I forgot the changelog once again... Added it!

**Comment by xFrednet** [comment]
@GuillaumeGomez before I r=blyxyas, could you run `cargo collect-metadata`? The previous bors CI complained about this check, if I saw it correctly

**Comment by GuillaumeGomez** [comment]
What message you want in the commit exactly? (So I can update it)

**Comment by xFrednet** [comment]
I wanted it to include links to the config and lint names. Usually we update the messages in the end, when the changelog is written, but some preprocessing never hurts ^^

**Comment by GuillaumeGomez** [comment]
Ran the command and updated the commit message.

**Comment by xFrednet** [comment]
Thank you for the quick updates :D

@bors r=blyxyas

**Pull request merged**